### PR TITLE
Task 3 Authorization

### PR DESCRIPTION
**Pull request by manaf92**
Add authorization to controllers and pages

### CODE REVIEW

**Review by OlSarLang — CHANGES_REQUESTED** [review]
This is not all that has to be done. We need authorize for Teachers only.

**Review by OlSarLang — CHANGES_REQUESTED** [review]
I can still access Users and Courses as a student. That should not be allowed.

**Review by manaf92 — COMMENTED** [review]
I Added authorized. it doesn't allow a student  access to create a course or a user

**Review by manaf92 — COMMENTED** [review]
Added authorize. removed Privacy

File: Views/Shared/_Layout.cshtml [hunk 1]
```diff
@@ -23,14 +23,17 @@
...
                         <li class="nav-item">
-                            <a class="nav-link text-dark" asp-area="" asp-controller="Home" asp-action="Privacy">Privacy</a>
+                            @if (User.IsInRole("Teacher"))
+                            {
+                                <a class="nav-link text-dark" asp-area="" asp-controller="SystemUser" asp-action="Index">Users</a>
+                            }
                         </li>
                         <li class="nav-item">
-                            <a class="nav-link text-dark" asp-area="" asp-controller="SystemUser" asp-action="Index">Users</a>
-                        </li>
```
Comment (Rocangus) [inline Views/Shared/_Layout.cshtml:30]: We still need to show teachers the list of users.

**Review by Rocangus — CHANGES_REQUESTED** [review]
Almost there. Just make sure that someone in the role Teacher can still access the list of users from the nav bar at the top of the page. Please refer to my comment and ask me if anything is unclear.

File: Views/Shared/_Layout.cshtml (hunk 1 above)
Comment (Rocangus) [reply Views/Shared/_Layout.cshtml:30]: My mistake, I misinterpreted the way the changes were presented. This is already done.

**Review by Rocangus — APPROVED** [review]
This looks good! I will merge now.

**Pull request merged**